File: src/features/typing/data/ts/partition_equal_subset_sum.ts
function canPartition(nums: number[]): boolean {
    let sum = 0;
    for (const num of nums) {
        sum += num;
    }
    if (sum % 2 === 1) {
        return false;
    }
    const target = sum / 2;
    const dp: boolean[] = new Array(target + 1).fill(false);
    dp[0] = true;
    for (const num of nums) {
        for (let j = target; j >= num; j--) {
            if (dp[j - num]) {
                dp[j] = true;
            }
        }
    }
    return dp[target];
}

File: src/features/typing/data/dart/partition_equal_subset_sum.dart
class Solution {
  bool canPartition(List<int> nums) {
    var sum = 0;
    for (final n in nums) {
      sum += n;
    }
    if (sum % 2 == 1) {
      return false;
    }
    final target = sum ~/ 2;
    final dp = List<bool>.filled(target + 1, false);
    dp[0] = true;
    for (final n in nums) {
      for (int j = target; j >= n; j--) {
        if (dp[j - n]) {
          dp[j] = true;
        }
      }
    }
    return dp[target];
  }
}
